core-customize/hybris/bin/custom/sapcxtools/sapcommercetoolkit/testsrc/tools/sapcx/commerce/toolkit/testing/testdoubles/user/UserServiceFake.java

```diff
@@ -20,12 +25,82 @@
 import tools.sapcx.commerce.toolkit.testing.itemmodel.InMemoryModelFactory;
 
 public class UserServiceFake implements UserService {
+
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	public static class Builder {
+		UserServiceFake fake = new UserServiceFake();
+
+		public Builder registerCurrentUser(UserModel user) {
+			fake.current = user;
+			fake.users.put(user.getUid(), user);
+			return this;
+		}
+
+		public Builder registerCurrentUserId(String customerId) {
+			CustomerModel customer = InMemoryModelFactory.createTestableItemModel(CustomerModel.class);
+			customer.setUid(customerId);
+			return registerCurrentUser(customer);
+		}
+
+		public Builder registerCurrentUserEntity(String customerId, Class<? extends CustomerModel> customerClass) {
+			CustomerModel customer = InMemoryModelFactory.createTestableItemModel(customerClass);
+			customer.setUid(customerId);
+			return registerCurrentUser(customer);
+		}
+
+		public Builder registerUserWithGroups(UserModel user, List<UserGroupModel> userGroups) {
+			linkUserAndGroupsBidirectional(user, userGroups);
+			userGroups.forEach(userGroup -> fake.userGroups.put(userGroup.getUid(), userGroup));
+			return registerCurrentUser(user);
+		}
+
+		private static void linkUserAndGroupsBidirectional(UserModel user, List<UserGroupModel> userGroups) {
+			user.setGroups(new HashSet<>(userGroups));
+			for (UserGroupModel userGroup : userGroups) {
+
+				Set<PrincipalModel> members;
+				if (userGroup.getMembers() == null || userGroup.getMembers().isEmpty()) {
+					members = new HashSet<>();
+				} else {
+					members = userGroup.getMembers();
+				}
+
+				members.add(user);
+				userGroup.setMembers(members);
+			}
+		}
+
+		public Builder registerGroups(List<UserGroupModel> userGroups) {
+			userGroups.forEach(userGroup -> fake.userGroups.put(userGroup.getUid(), userGroup));
+			return this;
+		}
+
+		public UserServiceFake build() {
+			EmployeeModel admin = InMemoryModelFactory.createTestableItemModel(EmployeeModel.class);
+			admin.setUid("admin");
+			fake.admin = admin;
```

Comment:
the admin user also needs to be added to all users within the fake, see the constructor of the Fake.